File: lib-segment/src/main/java/websiteschema/mpsegment/hmm/Node.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package websiteschema.mpsegment.hmm;

import java.io.IOException;
import websiteschema.mpsegment.util.ISerialize;
import websiteschema.mpsegment.util.SerializeHandler;

/**
 * Graphic Model Node
 * @author ray
 */
public class Node implements ISerialize {

    private int index = -1;
    private String name = null;

    public Node() {
    }
    
    public Node(String name) {
        this.name = name;
    }

    public Node(String name, int index) {
        this.name = name;
        this.index = index;
    }

    public int getIndex() {
        return index;
    }

    public void setIndex(int index) {
        this.index = index;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public void save(SerializeHandler writeHandler) throws IOException {
        writeHandler.serializeString(name);
        writeHandler.serializeInt(index);
    }

    @Override
    public void load(SerializeHandler readHandler) throws IOException {
        name = readHandler.deserializeString();
        index = readHandler.deserializeInt();
    }
}

File: lib-segment/src/main/scala/me/yingrui/segment/hmm/Node.scala
package me.yingrui.segment.hmm

import me.yingrui.segment.util.ISerialize
import me.yingrui.segment.util.SerializeHandler

class Node extends ISerialize {

  private var index = -1
  private var name: String = null

  def getIndex(): Int =  index

  def setIndex(index: Int) {
    this.index = index
  }

  def getName(): String = name

  def setName(name: String) {
    this.name = name
  }

  override def save(writeHandler: SerializeHandler) {
    writeHandler.serializeString(name)
    writeHandler.serializeInt(index)
  }

  override def load(readHandler: SerializeHandler) {
    name = readHandler.deserializeString()
    index = readHandler.deserializeInt()
  }
}

object Node {
  def apply() = new Node()

  def apply(name: String) = {
    val node = new Node()
    node.name = name
    node
  }

  def apply(name: String, index: Int) = {
    val node = new Node()
    node.name = name
    node.index = index
    node
  }
}
